### PR TITLE
CMake fixes for netCDF

### PR DESCRIPTION
**Pull request by johnomotani**
1. Linking to the github.com master-branch version of netCDF-cxx4 seems to require backporting a fix from `next` #2455.
2. Also, using the master-branch version makes the output of `ncxx4-config --version` be `4.3.2-developer`. CMake doesn't like the `-developer` suffix, so update the regex in `FindnetCDFCxx.cmake` to remove it.
3. Remove propagation of `NCXX_BINARY_DIR` in `bout++Config.cmake.in` since this variable seems to never be used.
4. `set()` the variables `NC_CONFIG` and `NCXX4_CONFIG`, if they exist. This allows packages that import BOUT++ to make use of these variables when finding netCDF and netCDF-cxx4, without having to pass them again.

### CODE REVIEW

**Comment by johnomotani** [comment]
@ZedThree this PR seems to fix the problems I was posting about on Slack yesterday. At least, I've managed to compile STORM against this branch.

I'm not at all sure about point 3 in the description above - I couldn't see `NCXX_BINARY_DIR` being used or set anywhere, but I'm still a CMake newbie, so might well have missed something.

I did also try to use `-DBOUT_DOWNLOAD_NETCDF_CXX4=ON`, but it was trying to link to the (ancient) system HDF5, not the HDF5 that netCDF-c was linked to. As I have something working now, I'm not going to chase that failure up.

**Comment by ZedThree** [comment]
I have a feeling `NCXX_BINARY_DIR` might be used for `BOUT_DOWNLOAD_NETCDF_CXX4` -- I'll investigate

File: bout++Config.cmake.in [hunk 1]
```diff
@@ -57,10 +57,15 @@ endif()
...
+  set(netCDF_ROOT "@netCDF_ROOT@")
+elseif(EXISTS "@NC_CONFIG@")
+  set(NC_CONFIG "@NC_CONFIG@")
+endif()
 if(EXISTS "@netCDFCxx_ROOT@")
   set(netCDFCxx_ROOT "@netCDFCxx_ROOT@")
-elseif(EXISTS "@NCXX_BINARY_DIR@")
-  list(APPEND CMAKE_PREFIX_PATH "@NCXX_BINARY_DIR@")
+elseif(EXISTS "@NCXX4_CONFIG@")
+  set(NCXX4_CONFIG "@NCXX4_CONFIG@")
```
Comment (ZedThree) [inline bout++Config.cmake.in:68]: Please restore these two lines, perhaps with the added comment:

```suggestion
  set(NCXX4_CONFIG "@NCXX4_CONFIG@")
elseif(EXISTS "@NCXX_BINARY_DIR@")
  # If we downloaded netCDF-cxx4, then we need to add its build directory to our search paths
  list(APPEND CMAKE_PREFIX_PATH "@NCXX_BINARY_DIR@")  
```

**Review by ZedThree — CHANGES_REQUESTED** [review]
LGTM, but the removed `NCXX_BINARY_DIR` branch is needed for `BOUT_DOWNLOAD_NETCDF_CX4`

**Review by ZedThree — APPROVED** [review]
Lovely, thanks @johnomotani !

**Pull request merged**